### PR TITLE
Check secure boot in TPM event log plugin

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
The first commit lgtm, the second will review later.

File: plugins/uefi/fu-self-test.c [hunk 1]
```diff
@@ -167,7 +167,7 @@ fu_uefi_device_func (void)
 }
 
 static void
-fu_uefi_vars_func (void)
```
Comment (hughsie) [inline plugins/uefi/fu-self-test.c]: This should move to libfwupdplugin if it's shared code now.

File: plugins/tpm-eventlog/fu-plugin-tpm-eventlog.c [hunk 2]
```diff
@@ -31,6 +32,30 @@ fu_plugin_destroy (FuPlugin *plugin)
...
+fu_plugin_tpm_eventlog_create_dummy (FuPlugin *plugin, const gchar *reason, GError **error)
+{
+	const gchar *key;
+	g_autoptr(FuDevice) dev = fu_device_new ();
+
+	key = fu_plugin_get_dmi_value (plugin, FU_HWIDS_KEY_MANUFACTURER);
+	if (key != NULL)
+		fu_device_set_vendor (dev, key);
+	key = "TPM measurements";
+	fu_device_set_name (dev, key);
```
Comment (hughsie) [inline plugins/tpm-eventlog/fu-plugin-tpm-eventlog.c]: not sure i see the reason for re-using `key`

File: plugins/tpm-eventlog/fu-plugin-tpm-eventlog.c [hunk 3]
```diff
@@ -31,6 +32,30 @@ fu_plugin_destroy (FuPlugin *plugin)
...
 
+static gboolean
+fu_plugin_tpm_eventlog_create_dummy (FuPlugin *plugin, const gchar *reason, GError **error)
+{
+	const gchar *key;
+	g_autoptr(FuDevice) dev = fu_device_new ();
+
+	key = fu_plugin_get_dmi_value (plugin, FU_HWIDS_KEY_MANUFACTURER);
+	if (key != NULL)
+		fu_device_set_vendor (dev, key);
```
Comment (hughsie) [inline plugins/tpm-eventlog/fu-plugin-tpm-eventlog.c]: We need a vendor_id too, no?

Comment (superm1) [reply plugins/tpm-eventlog/fu-plugin-tpm-eventlog.c]: The dummy device from UEFI it's modeled on didn't have one.  If you think so, then yes we should put it in both places.

**Comment by superm1** [comment]
@hughsie re-review this one?

**Comment by hughsie** [comment]
> re-review this one

Let me ponder, this all seems a bit magic. I'm not loving the dummy devices popping up everywhere, and I'm wondering if we want something more generic.

**Comment by hughsie** [comment]
> put it in UpdateMessage for the UEFI device

I think I'd prefer that for now; I'm pondering another thing to represent the "platform" and it's not all crystallised in my brain yet.

**Comment by superm1** [comment]
OK here's how it looks with that set of changes.

```
├─System Firmware:
│     Device ID:           6c24a747f97668873b761558e322398a91dbf394
│     Current version:     1.9.1
│     Minimum Version:     1.9.1
│     Vendor:              Dell Inc. (DMI:Dell Inc.)
│     Update Message:      Platform firmware measurement unavailable. Secure boot is disabled in BIOS setup, enabling it may fix this issue
│     GUID:                ce945437-7358-49f1-95d8-6b694a10a755
│     Device Flags:        • Internal device
│                          • Updatable
│                          • Requires AC power
│                          • Supported on remote server
│                          • Needs a reboot after installation
│                          • Cryptographic hash verification is available
│                          • Device is usable for the duration of the update
```

**Pull request merged**